lib/DXIL/DxilMetadataHelper.cpp

```diff
@@ -617,9 +617,11 @@ void DxilMDHelper::EmitDxilResourceBase(const DxilResourceBase &R, Metadata *ppM
   Constant *GlobalSymbol = R.GetGlobalSymbol();
   // For sm66+, global symbol will be mutated into handle type.
   // Save hlsl type by generate bitcast on global symbol.
```

Comment:
adding a line to explain the creation of an HLSL undef for the global symbol would be helpful since it is mingled with what this comment describes.